extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-mysql/src/main/java/org/apache/guacamole/auth/mysql/MySQLAuthenticationProviderModule.java

```diff
@@ -58,11 +59,20 @@ public MySQLAuthenticationProviderModule(MySQLEnvironment environment)
 
         // Set the MySQL-specific properties for MyBatis.
         myBatisProperties.setProperty("mybatis.environment.id", "guacamole");
-        myBatisProperties.setProperty("JDBC.host", environment.getMySQLHostname());
-        myBatisProperties.setProperty("JDBC.port", String.valueOf(environment.getMySQLPort()));
-        myBatisProperties.setProperty("JDBC.schema", environment.getMySQLDatabase());
-        myBatisProperties.setProperty("JDBC.username", environment.getMySQLUsername());
-        myBatisProperties.setProperty("JDBC.password", environment.getMySQLPassword());
+        
+        // Check for URI
+        URI mySQLUri = environment.getMySQLUri();
```

Comment:
This is likely a good change to have, allowing users to override the entire JDBC URL, but it seems out of scope for the nature of this particular change. The [GUACAMOLE-678](https://issues.apache.org/jira/browse/GUACAMOLE-678) issue is pretty specifically about defining a new property type:

> Implement a new property type, UriGuacamoleProperty, that takes a string and generates a URI object, validating the URI in the process. With the increased number of extensions that take URIs as configuration parameters this will allow for more central validation of the URI before it gets used by the module.

I agree that it makes sense to refactor existing code to migrate to that type wherever appropriate, but this particular change adds a new feature that deserves its own issue.

Reply:
Yeah, you're probably right - I'll refactor and pull this code out, and open a separate JIRA issue for it.

Reply:
Okay, JDBC URI configuration has been extracted and will be moved to GUACAMOLE-766.